restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1008,13 +1022,16 @@ private ExecutionContext nestedPopulate(ExecutionContext context) {
                         .from(keyspace, collection)
                         .where(keyCondition)
                         .where(buildConditions(nestedCondition, db.treatBooleansAsNumeric()))
+                        .limit(1)
                         .allowFiltering()
                         .build()
                         .bind();
 
                 // If the nested query finds any docs, return the input doc to preserve
-                // the paging order of the main query
-                return db.executeSelect(nestedQuery, 1, null, nestedCtx).take(1).map(nested -> d);
+                // the paging order of the main query.
+                // Note: use page size larger than the query limit to make sure the backend
```

Comment:
Can you maybe explain me what the page size actually means in this concept? I am not sure if I follow 100% here. And why the limit fix is not good enough?

Reply:
This page size maps to the C* result set page size. 

Our Rx pipeline may have to process multiple pages, however C* will ignore paging state if page _size_ is not set. Therefore, we require page size to be set up front because we _may_ have to use it to request page #2, 3, etc...

The `LIMIT 1` is the real fix that makes sure we do not request more than 1 page for "filter" queries. We only care whether we have any rows or no rows. Previously we requested page size `1` for these queries without `LIMIT`, but in this case C* does not always know whether more matching rows are available and will return a non-null paging state, which in turn, will cause our Rx pipeline to fetch page #2 only to reliase there are no more rows available. With `LIMIT 1` and page size 2 (> 1) we can be sure that the first page will not have a paging state. In C* terms the result set will be "exhausted".

Hypothetically, we might do fine without any page size in this case, but as noted above, in the general case page size is required, so I thought it would be ok just to use page size == `2` here.